### PR TITLE
[io] Inline functions to avoid multiple definition error

### PR DESCRIPTION
**Pull request by brunoabinader**

### CODE REVIEW

File: mapbox/io/include/mapbox/io.hpp [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace base {
 using ErrorType = std::string;
 
 // NOLINTNEXTLINE(misc-definitions-in-headers)
```
Comment (tmpsantos) [inline mapbox/io/include/mapbox/io.hpp]: If you make this inline you can remove this `NOLINT`

**Review by tmpsantos — APPROVED** [review]
LGTM with a nit

File: mapbox/io/include/mapbox/io.hpp [hunk 2]
```diff
@@ -15,8 +15,7 @@ namespace base {
 
 using ErrorType = std::string;
 
-// NOLINTNEXTLINE(misc-definitions-in-headers)
-nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
+inline nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
```
Comment (pozdnyakov) [inline mapbox/io/include/mapbox/io.hpp:18]: maybe better to add definition in a .cpp file?

Comment (brunoabinader) [reply mapbox/io/include/mapbox/io.hpp:18]: I thought about it - but it'd break the header-only nature of those libraries, no?

Comment (pozdnyakov) [reply mapbox/io/include/mapbox/io.hpp:18]: It's not a template class - imo does not have to be header-only.. 

Also, inline functions might bring some problems, e.g. someone could define
```
inline nonstd::expected<std::string, ErrorType> readFile(const ghc::filesystem::path& filename) {
// Unexpected leftover implementation here, or maybe previous implementation
}
```
Both implementations will build fine together, then arbitrary version will be picked for execution (if inline failed)..

Comment (brunoabinader) [reply mapbox/io/include/mapbox/io.hpp:18]: :point_right: https://github.com/mapbox/mapbox-base/pull/24

**Pull request merged**